src/Compilers/VisualBasic/Portable/Binding/SourceFileBinder.vb

```diff
@@ -81,6 +81,17 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             End Get
         End Property
 
+        Public Overrides ReadOnly Property CheckOverflow As Boolean
+            Get
+                ' VBCore should always be compiled with overflow checks on.
+                If SyntaxTree.GetEmbeddedKind() = EmbeddedSymbolKind.VbCore Then
```

Comment:
Since the newly added test is expected to be less likely than the existing test (most code is user code), the order of these two tests should probably be swapped.